src/main/scala/com/fulcrumgenomics/bam/FlagStat.scala

```diff
@@ -0,0 +1,161 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.FgBioDef.PathToBam
+import com.fulcrumgenomics.bam.api.{SamRecord, SamSource}
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.CommonsDef.SafelyClosable
+import com.fulcrumgenomics.commons.util.LazyLogging
+import com.fulcrumgenomics.sopt.{arg, clp}
+import com.fulcrumgenomics.util.Metric
+
+/** Stores various metrics derived from the FLAG field in a SAM or BAM file.  The metric counts either reads that
+  * pass-filter or reads that do not pass filter, but not both.
+  *
+  * See the [SAMTools flagstat documentation](http://www.htslib.org/doc/samtools.html)
+  *
+  * @param pf true if the metric is for pass-filter reads, false for reads that did not pass-filter.
+  * @param reads the total number of reads/records.
+  * @param mapped the number of mapped reads/records.
+  * @param paired the number of reads/records that were paired.
+  * @param mapped_in_pair the number of mapped paired reads/records with its mate mapped.
+  * @param proper_paired the number of paired reads/records that were properly paired.
+  * @param singleton the number of paired reads/records that were mapped but its mate was not mapped.
+  * @param read_one the number of first of pair reads/records.
+  * @param read_two the number of second of pair reads/records.
+  * @param duplicate the number of duplicate reads/records.
+  * @param inter_chrom the number of mapped paired reads/records whose mate maps to a different chromosome.
+  * @param inter_chrom_mapq5 the number of mapped paired reads/records with mapping quality at least five whose mate maps to a different chromosome.
+  * @param secondary the number of secondary reads/records.
+  * @param supplementary the number of supplementary reads/records.
+  */
+case class FlagStatMetric
+(pf: Boolean,
+ var reads: Long = 0,
+ var mapped: Long = 0,
+ var paired: Long = 0,
+ var mapped_in_pair: Long = 0,
+ var proper_paired: Long = 0,
+ var singleton: Long = 0,
+ var read_one: Long = 0,
+ var read_two: Long = 0,
+ var duplicate: Long = 0,
+ var inter_chrom: Long = 0,
+ var inter_chrom_mapq5: Long = 0,
+ var secondary: Long = 0,
+ var supplementary: Long = 0
+) extends Metric
+
+@clp(group = ClpGroups.SamOrBam, description=
+  """|Collects metrics from the FLAG field in a SAM or BAM file.
+  """)
+class FlagStat
+( @arg(flag='i', doc="Input SAM or BAM file.") val input: PathToBam,
+  @arg(flag='o', doc="The output file in which metrics will be stored.") val output: Option[PathToBam] = None
```

Comment:
`PathToBam` -> `FilePath`.
Also maybe mention what happens if you don't provide it?